integration-tests/src/test/kotlin/com/google/devtools/ksp/test/ResourcesIT.kt

```diff
@@ -0,0 +1,74 @@
+package com.google.devtools.ksp.test
+
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.jetbrains.kotlin.incremental.deleteRecursivelyOrThrow
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class ResourcesIT(val useKSP2: Boolean) {
+    @Rule
+    @JvmField
+    val project: TemporaryTestProject = TemporaryTestProject("resources", useKSP2 = useKSP2)
+
+    @Test
+    fun testResources() {
+        val gradleRunner = GradleRunner.create().withProjectDir(project.root)
+        gradleRunner.withArguments(":workload:assemble").build().let { result ->
+            result.assertOutputContains("foo.txt: Hello")
+            result.assertOutputContains("foo/bar/baz/quux: World")
+        }
+    }
+
+    @Test
+    fun testUpToDate() {
```

Comment:
FYI this test will fail until Gradle is updated due to issues with how older Gradle handles `SkipWhenEmpty` and `ListProperty<Directory>`